### ISSUE TITLE
Update "MagicNumber" to not mark "Named Arguments" as code smells

### ISSUE THREAD

**Issue by DmytroTroynikov** [arturbosch/detekt#415]
Consider following example:

```kotlin
class HtlExpressionInsertHandler :
    HtlTextInsertHandler("=\"\${}\"", offset = 4)
```

`offset = 4` will be marked as "MagicNumber" rule violation.

The above technique is called [Named Arguments](https://kotlinlang.org/docs/reference/functions.html#named-arguments).
As you could see from the official documentation, the purpose of "Named Arguments" is:

* Improve convenience on caller side in case if called function has many parameters and/or parameters with default values
* Improve readability of code as argument names should explain the purpose of passed arguments (mostly for literals)

I think that current implementation of "MagicNumber" rule contradicts the second point, so it you should be updated as following:

"MagicNumber" passed via *Named Argument* should **not** be considered as code smell as the name of argument should explain the purpose of the number.

Thanks, 
Dmytro

**Comment by schalkms** [comment]
Thanks for sharing your thoughts!
I agree. Furthermore, I think this should be a configuration option which is set to _ignore=true_ per default.

**Comment by MyDogTom** [comment]
Glad to see, that this will be configurable. Because, I have different use case, where named argument is not enough.
```kotlin
class Example(private val adapter: StoriesAdapter){
    fun withMagic(){
        adapter.addStories(list = emptyList(), groupId = 0)
        adapter.addStories(list = emptyList(), groupId = 1)
    }

    fun noMagic(){
        adapter.addStories(list = emptyList(), groupId = FEED_STORIES_GROUP_ID)
        adapter.addStories(list = emptyList(), groupId = SUGGESTIONS_STORIES_GROUP_ID)
    }
}
```

**Comment by arturbosch** [comment]
Thanks for the comment, it's kind of the same reasoning as I stated in #455.

**Comment by Mauin** [comment]
This can be merged now as #426 was merged.
